controllers/cf/cf_pod_spec.go

```diff
@@ -0,0 +1,318 @@
+		(this.previousBasePodSpec == nil || !reflect.DeepEqual(this.basePodSpec, this.previousBasePodSpec))
+}
+
+func (this *PodSpecCF) Respond() {
+
+	if entry, exists := this.svcResourceCache.Get(resources.RC_KEY_DEPLOYMENT); exists {
+		entry.ApplyPatch(func(value interface{}) interface{} {
+			deployment := value.(*apps.Deployment).DeepCopy()
+
+			deployment.Spec.Template.Spec = *this.targetPodSpec
+
+			this.previousBasePodSpec = this.basePodSpec
+
+			return deployment
+		})
+	}
+}
+
+func (this *PodSpecCF) Cleanup() bool {
+	// No cleanup
+	return true
+}
+
+/*
+Not allowed:
+
+- affinity [alternative exists]
+- containers[*]. [only a single container without a name]
+  - containers.args [reserved]
+  - containers.command [reserved]
+  - containers.env [alternative exists]
+  - containers.image [reserved]
+  - containers.imagePullPolicy [alternative exists]
+  - containers.name [reserved]
+  - containers.ports [reserved]
+  - containers.workingDir [reserved]
+- ephemeralContainers [reserved]
+- imagePullSecrets [alternative exists]
+- initContainers [reserved]
+- tolerations [alternative exists]
+
+Rules:
+
+If you are setting a value of a field in the podSpec or podSpec.containers[0],
+this value must be valid as a whole.
+
+For example, when setting the initialDelaySeconds field of a readiness probe, you must also provide the probe handler.
+
+The operator may still modify the values you provided,
+but it will not add required sub-fields or fix an invalid value.
+*/
+func SanitizeBasePodSpec(log *zap.SugaredLogger, base *ar.ApicurioRegistrySpecDeploymentPodSpecPreview, current *core.PodSpec, factory *core.PodSpec) (*ar.ApicurioRegistrySpecDeploymentPodSpecPreview, *SanitizeError) {
+	// We are using values from *current* with fields (values):
+	//   - that the user cannot change
+	//   - or are empty by default
+	//   - or there is a CF that handles them
+	// We are using values from *factory* with fields (values):
+	//   - that are not empty by default, so empty fields in the base podSpec do not remove default values
+	//   - and there is not and existing CF that handles them
+	// Otherwise we just pass the base values along, relying on CFs to override things
+
+	base = base.DeepCopy()       // Defensive copy TODO is this needed?
+	current = current.DeepCopy() // Defensive copy TODO is this needed?
+	// affinity
+	if base.Affinity != nil {
+		return nil, NewConfigError("affinity")
+	}
+	base.Affinity = current.Affinity
+	// containers[*]
+	if len(base.Containers) > 1 {
+		return nil, NewConfigError("field containers must contain at most one item")
+
+	} else if len(base.Containers) == 1 {
+		baseContainer := &base.Containers[0]
+		// We have only a single container at the moment,
+		// TODO but in the future we should use the name field.
+		currentContainer := current.Containers[0]
+		factoryContainer := factory.Containers[0]
+
+		// containers[*].args
+		if len(baseContainer.Args) > 0 {
```

Comment:
please, help me understand the logic here:

`baseContainer` -> is the container described in the CR by the user?
`currentContainer` -> is the version of the container coming from the environment
`factoryContainer` -> this is the desired status assuming no `podTemplate` have been defined

correct?